src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -208,14 +211,14 @@ private unsafe bool HandleItems(ActionType actionType, QueryOperationContext que
 
                                 foreach (var referencedDocument in references)
                                 {
-                                    lastEtag = referencedDocument.Etag;
                                     hasChanges = true;
-                                    inMemoryStats.UpdateLastEtag(lastEtag, isTombstone);
 
-                                    var items = GetItemsFromCollectionThatReference(queryContext, indexContext, collection, referencedDocument, lastIndexedEtag, indexed);
+                                    var items = GetItemsFromCollectionThatReference(queryContext, indexContext, collection, referencedDocument, lastIndexedEtag, indexed, actionType);
 
                                     using (var itemsEnumerator = _index.GetMapEnumerator(items, collection, indexContext, collectionStats, _index.Type))
                                     {
+                                        var isFirst = true;
```

Comment:
Probably better to remove this and just state that the `CanContinue` will do a check every N items

Reply:
done